### PR TITLE
improve names docstring

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/3171

@mortenpi - could you have a quick look if this is correct 😄.
The relevant page in the manual is https://dataframes.juliadata.org/dev/lib/indexing/#General-rules
Thank you!

### CODE REVIEW

**Comment by mortenpi** [comment]
LGTM. I'd expect some warnings in the logs if they weren't working, and there doesn't appear to be any.

The easiest way would be to check if the references are fixed in the built docs though, for which you could set `push_preview = true` in `deploydocs` to get PR preview builds (from PRs from the same repo).

**Comment by bkamins** [comment]
Thank you!

File: src/abstractdataframe/abstractdataframe.jl [hunk 1]
```diff
@@ -39,7 +39,9 @@ Return a freshly allocated `Vector{String}` of names of columns contained in `df
 If `cols` is passed then restrict returned column names to those matching the
 selector (this is useful in particular with regular expressions, `Cols`, `Not`, and `Between`).
 `cols` can be:
-* any column selector ($COLUMNINDEX_STR; $MULTICOLUMNINDEX_STR)
+* any column selector ($COLUMNINDEX_STR; $MULTICOLUMNINDEX_STR); these column
+  selectors are documented in [General rules](@ref) section of the [Indexing](@ref)
+  part of the DataFrames.jl documentation
```
Comment (nalimilan) [inline src/abstractdataframe/abstractdataframe.jl:44]:
```suggestion
* any column selector ($COLUMNINDEX_STR; $MULTICOLUMNINDEX_STR); these column
  selectors are documented in the [General rules](@ref) section of the [Indexing](@ref)
  part of the DataFrames.jl manual
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Comment by bkamins** [comment]
All is OK on `main`.

**Pull request merged**